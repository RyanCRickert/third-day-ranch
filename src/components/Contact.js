import React from "react";

const Contact = () => (
  <div className="landing">
    Contact Us
  </div>
);

export default Contact;