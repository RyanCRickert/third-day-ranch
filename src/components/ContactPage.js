import React from "react";

const ContactPage = () => (
  <div className="contact">
    <div className="contact-body">
      <div>Give us a call to see what events we currently have going on!</div>
      <div>409-291-1456</div>
      <div>or come by and see us at:</div>
      <div>373 Country Road 092, Jasper, TX 75951</div>
    </div>
  </div>
);

export default ContactPage;