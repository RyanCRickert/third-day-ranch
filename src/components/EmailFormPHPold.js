import React from "react";
import { Form, FormGroup, FormText, Input, Label, Button } from "reactstrap";
import { post } from "axios";

import ThankYouModal from "./ThankYouModal";

export default class EmailFormPHP extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      email: "",
      message: "",
      name: "",
      file: undefined
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleFileChange = this.handleFileChange.bind(this);
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  handleFileChange(e) {
    this.setState({ file: e.target.files[0] })
  }

  handleSubmit(e) {
    e.preventDefault();
    var formData = new FormData();

    if (this.state.file != undefined) {
      formData.append('file', this.state.file);
    }
    formData.append("name", this.state.name);
    formData.append("email", this.state.email);
    formData.append("message", this.state.message);
    const config = {
      headers: {
          'content-type': 'multipart/form-data'
      }
    }

    return post("http://neuraldynamicstechnologies.com/phpmailer.php", formData, config).then(function (response) {
      console.log(response)
    });
  }

  render() {
    return (
      <div>
        <ThankYouModal message={this.props.message} closeThankYouModal={this.closeThankYouModal} submitted={this.state.submitted} />
        <Form onSubmit={this.handleSubmit}>
          <FormGroup>
            <Label for="name" className="form__label">Name</Label>
            <Input
              type="text"
              name="name"
              className="form__name"
              onChange={this.handleChange} />
          </FormGroup>
          <FormGroup>
            <Label for="email" className="form__label">Email</Label>
            <Input
              type="email"
              name="email"
              className="form__email"
              onChange={this.handleChange} />
          </FormGroup>
          <FormGroup>
            <Label for="message" className="form__label">Message</Label>
            <Input
              type="textarea"
              name="message"
              className="form__message"
              value={this.state.message}
              onChange={this.handleChange} />
          </FormGroup>
          <FormGroup>
            <Label for="file" className="form__label">File</Label>
            <Input
              type="file"
              name="file"
              onChange={this.handleFileChange} 
              className="form__file-input"/>
            <FormText color="muted" className="form__text">
              File must be sent as PDF, DOCX or TXT.
            </FormText>
          </FormGroup>
          <Button disabled={this.state.name == 0 || this.state.email == 0 || this.state.message == 0 } className="form__button">
            {this.state.name == 0 || this.state.email == 0 || this.state.message == 0 ? "Please fill out each form" : "Submit"}
          </Button>
        </Form>
        {this.state.verification && <p>{this.state.verification}</p>}
      </div>
    )
  }
}