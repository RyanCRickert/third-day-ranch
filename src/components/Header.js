import React from "react";
import { Link } from "react-router-dom";
import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';

export default class Header extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      width: 0,
      height: 0,
      dropdownOpen: false
    };

    this.updateWindowDimensions = this.updateWindowDimensions.bind(this);
    this.toggle = this.toggle.bind(this);
  }

  componentDidMount() {
    this.updateWindowDimensions();
    window.addEventListener('resize', this.updateWindowDimensions);
  }
  
  componentWillUnmount() {
    window.removeEventListener('resize', this.updateWindowDimensions);
  }
  
  updateWindowDimensions() {
    this.setState({ width: window.innerWidth, height: window.innerHeight });
  }

  toggle() {
    this.setState(prevState => ({
      dropdownOpen: !prevState.dropdownOpen
    }));
  }

  render() {
    return (
      <header className="header">
        <div className="content-container__header">
          <div className="header__content">
            <div>
            <Link to="/" style={{ textDecoration: 'none' }}>
              <div className={this.state.width >= 830 ? "header__title" : "header__title header-shrink"}>
              {this.state.width >= 830 ? <h1><img className="header__image" src="/images/logo-only.png" /><div className="header__company">Neural Dynamics Technologies LLC</div></h1> : <div className="header-shrunk"><img className="header__image" src="/images/logo-only.png" /><h1><div>Neural</div><div>Dynamics</div><div>Technologies LLC</div></h1></div>}
              </div>
            </Link>
            </div>
            {this.state.width > 700 ?
            <div className="header__buttons">
              <Link to="/" className="header__link" style={{ textDecoration: 'none' }}>
                HOME
              </Link>
              <Link to="/about" className="header__link" style={{ textDecoration: 'none' }}>
                ABOUT
              </Link>
              <Link to="/technology" className="header__link" style={{ textDecoration: 'none' }}>
                TECHNOLOGY
              </Link>
              <Link to="/contact" className="header__link" style={{ textDecoration: 'none' }}>
                CONTACT
              </Link>
            </div>
            :
              <Dropdown isOpen={this.state.dropdownOpen} toggle={this.toggle}>
              <DropdownToggle caret className="header__dropdown">
                Navigation
              </DropdownToggle>
              <DropdownMenu style={{ zIndex: 1002}}>
                <DropdownItem>
                  <Link to="/" className="header__link" style={{ textDecoration: 'none' }}>
                  HOME
                  </Link>
                </DropdownItem>
                <DropdownItem>
                  <Link to="/about" className="header__link" style={{ textDecoration: 'none' }}>
                  ABOUT
                  </Link>
                </DropdownItem>
                <DropdownItem>
                  <Link to="/technology" className="header__link" style={{ textDecoration: 'none' }}>
                  TECHNOLOGY
                  </Link>
                </DropdownItem>
                <DropdownItem>
                  <Link to="/contact" className="header__link" style={{ textDecoration: 'none' }}>
                  CONTACT
                  </Link>
                </DropdownItem>
              </DropdownMenu>
              </Dropdown>
            }

          </div>
        </div>
      </header>
    )
  }
}