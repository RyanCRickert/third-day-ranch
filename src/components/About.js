import React from "react";
import { Form, FormGroup, Input, Label, Button } from "reactstrap";

import Header from "./Header";
import Footer from "./Footer";

export default class HomePage extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div><Header />
        <div className="about">
        </div>
        <div className="about__points">
          <div className="about__points-each">At Neural Dynamics Technologies (NDT), we design and manufacture neural interfaces for a variety of applications, ranging from tools for basic neuroscience research to therapeutic devices.</div>
          <div className="about__points-each">We take an interdisciplinary approach by working closely together with academics and clinicians to develop the best tools for elucidating brain function and for treating neurological disorders.</div>
          <div className="about__points-each">NDT was started in 2018 by a team of neuroscientists and engineers that were working together at MIT and is supported by the NIH.</div>
        </div>
      </div>
    )
  }
};