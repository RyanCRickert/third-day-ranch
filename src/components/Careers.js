import React from "react";
import { Collapse, Button, CardBody, Card } from 'reactstrap';

import Header from "./Header";
import Footer from "./Footer";
import EmailFormPHP from "./EmailFormPHP";
import FontAwesome from "react-fontawesome";

const contactMessage = "Thank you for taking the time to send us a message, we will view it as soon as we can and respond to you quickly.";

const applyMessage = "Thank you for your interest in Neural Dynamics Technologies.  We will review you application and should you be a good match we will contact you.";

export default class Contact extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      collapseContact: true
    }

    this.toggleContact = this.toggleContact.bind(this);
    this.toggleMicrofab = this.toggleMicrofab.bind(this);
  }

  toggleContact() {
    this.setState({ collapseContact: !this.state.collapseContact });
  }

  toggleMicrofab() {
    this.setState({ collapseMicroFab: !this.state.collapseMicroFab });
  }

  render() {
    return (
      <div><Header />
        <div className="career-container career__points">
          <Button onClick={this.toggleContact} className="career__button-contact"><div className="career__button-90">Contact Us</div><div className="career__button-10"><FontAwesome className={this.state.collapseContact ? "career__flipped" : "career__noflip"} name="caret-down"/></div></Button>
          <Collapse isOpen={this.state.collapseContact}>
            <Card>
              <CardBody>
                <div className="career__explain">
                Please feel free to get in touch with questions on our products or on partnering, or if you are interested in career opportunities.<br/>
                <div className="career__separator" />
                </div>
                <EmailFormPHP message={contactMessage}/>
              </CardBody>
            </Card>
          </Collapse>
        </div>
      </div>
    )
  }
};