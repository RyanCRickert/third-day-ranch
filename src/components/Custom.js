import React from "react";

import Header from "./Header";
import Footer from "./Footer";

export default class Custom extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div><Header />
        <div className="custom">
        </div>
        <div className="content-container custom__points">
          Our design / our work / publications which would make this a different section?
        </div>
      </div>
    )
  }
}