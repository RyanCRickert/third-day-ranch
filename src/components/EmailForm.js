import React from "react";
import { Form, FormGroup, FormText, Input, Label, Button } from "reactstrap";
import { post } from "axios";

import ThankYouModal from "./ThankYouModal";
import Header from "./Header";
import Footer from "./Footer";

export default class Contact extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      name: "",
      email: "",
      message: "",
      file: undefined,
      submitted: false,
    }

    this.handleChange = this.handleChange.bind(this);
    this.handleFileChange = this.handleFileChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.closeThankYouModal = this.closeThankYouModal.bind(this);
    this.fileUpload = this.fileUpload.bind(this);
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  handleFileChange(e) {
    this.setState({ file: e.target.files[0] })
  }

  handleSubmit(e) {
    e.preventDefault();
    const { name, email, message } = this.state;

    this.setState({ submitted: true, message: "" });
    e.target.message = "";


    if(this.state.file != undefined) {
      this.fileUpload(this.state).then((response)=>{
        console.log(response.data);
      })
    } else {
      return post("/api/form",{
        name,
        email,
        message
      })
    }
  }

  fileUpload(state){
    const url = "/api/formPDF";
    const formData = new FormData();

    formData.append('file', state.file);
    formData.set("name", this.state.name);
    formData.set("email", this.state.email);
    formData.set("message", this.state.message);
    const config = {
        headers: {
            'content-type': 'multipart/form-data'
        }
    }
    return  post(url, formData, config)
  }

  closeThankYouModal() {
    this.setState({ submitted: false });
  }

  render() {
    return (
      <div>
        <ThankYouModal message={this.props.message} closeThankYouModal={this.closeThankYouModal} submitted={this.state.submitted} />
        <Form onSubmit={this.handleSubmit}>
          <FormGroup>
            <Label for="name" className="form__label">Name</Label>
            <Input
              type="text"
              name="name"
              className="form__name"
              onChange={this.handleChange} />
          </FormGroup>
          <FormGroup>
            <Label for="email" className="form__label">Email</Label>
            <Input
              type="email"
              name="email"
              className="form__email"
              onChange={this.handleChange} />
          </FormGroup>
          <FormGroup>
            <Label for="message" className="form__label">Message</Label>
            <Input
              type="textarea"
              name="message"
              className="form__message"
              value={this.state.message}
              onChange={this.handleChange} />
          </FormGroup>
          <FormGroup>
            <Label for="file" className="form__label">File</Label>
            <Input
              type="file"
              name="file"
              onChange={this.handleFileChange} 
              className="form__file-input"/>
            <FormText color="muted" className="form__text">
              File must be sent as PDF, DOCX or TXT.
            </FormText>
          </FormGroup>
          <Button disabled={this.state.name == 0 || this.state.email == 0 || this.state.message == 0 } className="form__button">
            {this.state.name == 0 || this.state.email == 0 || this.state.message == 0 ? "Please fill out each field" : "Submit"}
          </Button>
        </Form>
      </div>
    )
  }
};