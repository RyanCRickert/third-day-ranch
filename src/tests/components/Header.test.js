import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { Header } from "../../components/Header";

let startLogout, wrapper;

beforeEach (() => {
  startLogout = jest.fn();
  wrapper = shallow(<Header
    startLogout={startLogout}
    />);
});

test("should render Header correctly", () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test("should call startLogout on button click", () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
  wrapper.find("button").simulate("click")
  expect(startLogout).toHaveBeenCalled();
  expect(toJSON(wrapper)).toMatchSnapshot();
});