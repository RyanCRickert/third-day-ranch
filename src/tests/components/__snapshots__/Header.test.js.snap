// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call startLogout on button click 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Boilerplate
        </h1>
      </Link>
      <button
        className="button button--link"
        onClick={[MockFunction]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;

exports[`should call startLogout on button click 2`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Boilerplate
        </h1>
      </Link>
      <button
        className="button button--link"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
          }
        }
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Boilerplate
        </h1>
      </Link>
      <button
        className="button button--link"
        onClick={[MockFunction]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
